helm/cluster-vsphere/templates/ipam/assign-ip-pre-install-job.yaml

```diff
@@ -79,9 +75,10 @@ spec:
       securityContext:
         runAsUser: {{ include "securityContext.runAsUser" $ }}
         runAsGroup: {{ include "securityContext.runAsGroup" $ }}
-      containers:
-        - name: post-install-job
-          image: "{{ .Values.kubectlImage.registry }}/{{ .Values.kubectlImage.name }}:{{ .Values.kubectlImage.tag }}"
+      initContainers:
```

Comment:
Asking to understand your choices here. Why did you use initContainers here? Is it to ensure this will be run first? Why didn't you use a single container like you did above in pre hooks?

Reply:
it would look even more like spaghetti having the control logic from helm in the middle of bash ;)
![Screenshot 2023-10-26 at 10 25 47](https://github.com/giantswarm/cluster-vsphere/assets/535866/8d8cc525-ea94-429a-a747-05c7de691706)

I think this way the concerns are better separated and also the logs can be shown just for each feature separately. The slowdown here is negligible, it's the same container image all the time. That's acutally a way how Tekton (internally) also serialises their steps or at least used to do that this way back then.

Reply:
I am satisfied with this answer :D